### DISCUSSION TITLE
no module named 'aioespnow' on esp8266 4MB  v1.21.0

### DISCUSSION THREAD

**Discussion by Fisherman48** [micropython/micropython#12789] [ESP8266] [answered]
In the docs i can read:

> A supplementary module ([aioespnow](https://docs.micropython.org/en/latest/library/espnow.html#module-aioespnow)) is available to provide [asyncio](https://docs.micropython.org/en/latest/library/asyncio.html) support.
In the docs:
> Note: Asyncio support is available on all ESP32 targets as well as those ESP8266 boards which include the asyncio module (ie. ESP8266 devices with at least 2MB flash memory).

But there is no  'aioespnow' on two ESP8266 4MB flashed to  v1.21.0

`ImportError: no module named 'aioespnow'

MicroPython v1.21.0 on 2023-10-05; ESP module with ESP8266
Type "help()" for more information.

>>> import esp
>>> esp.flash_size()
4194304
>>> 

/Stefan

**Comment by sir-user** [comment]
Hi Stefan, 

Firstly, I am new to micropython but I have a suggestion. Could you check the manifest.py file? You might need to add it. 
Add this line: `require("aiospnow")`
I hope it helps.

Volkan

**Comment by jimmo** [comment, marked as answer]
@Fisherman48 Only the ESP32 includes `aioespnow` by default. On esp8266, you can install it manually (e.g. `mpremote mip install aioespnow`). The note in the documentation is to indicate that aioespnow is supported, not that it's included.

**Reply by jimmo** [reply]
(Or if you're building your own firmware, you can do what @sir-user suggests above, and include it in the manifest).

**Reply by Fisherman48** [reply]
@jimmo Ok, thanks.

After hours of trial and error using Win10, I can now see some "light in the tunnel", using CLI,  the power of mpremote and now have the aioespnow.mpy installed on esp8266.

Btw - I saw in an other thread about elaborating on difficulties in learning :
Depending of different learning styles - most of us prefer visual info like in GUI,  compared to reading/ writing text like CLI.

An ex that help me "completing the picture" : https://www.tomshardware.com/how-to/raspberry-pi-pico-micropython-mip

/Stefan
